driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java

```diff
@@ -687,8 +697,14 @@ public void sendSucceededEvent(final ResponseBuffers responseBuffers) {
                                                                 ? new BsonDocument()
                                                                 : getResponseDocument(responseBuffers, message.getId(),
                         new RawBsonDocumentCodec());
-                sendCommandSucceededEvent(message, commandName, responseDocumentForEvent, description, startTimeNanos,
-                        commandListener);
+                sendCommandSucceededEvent(message, commandName, responseDocumentForEvent, description, startTimeNanos, commandListener);
+            }
+        }
+
+        public void sendSucceededEventForOneWayCommand() {
+            if (commandListener != null && opened()) {
+                BsonDocument responseDocumentForEvent = new BsonDocument("ok", new BsonInt32(1));
```

Comment:
This is a bit of a layer violation:  the caller should really be responsible for deciding what the "faked" response to a command with no response looks like.  But as this and all the callers are not-public, I didn't think it was worth it to have the caller pass this in.   If you're unconvinced by that argument, the fix could be as simple as adding a new method to CommandMessage that returns this document.